platform/m6/sub/101a/stubs_entry_2.S

```diff
@@ -0,0 +1,60 @@
+#include "stubs_asm.h"
+
+#define NULL_SUB 0xE0020755
+
+// from Ghidra; each address needs to add 1 THUMB bit
+NHSTUB(CancelHPTimer,                       0xe0178e9f) // TODO: different method signature
+NHSTUB(DeleteDirectory_Fut,                 0xe0334fdd)
+NHSTUB(DeleteFile_Fut,                      0xe0334f93)
+NHSTUB(Feof_Fut,                            0xe0335189)
+NHSTUB(Fflush_Fut,                          0xe03351b3)
+NHSTUB(GetDrive_TotalClusters,              0xe003a195)
+NHSTUB(RenameFile_Fut,                      0xe0334f31)
+NHSTUB(SetFileAttributes,                   0xe035f66f)
+NHSTUB(SetFileTimeStamp,                    0xe035f627) // TODO: different method signature
+NHSTUB(SetHPTimerAfterNow,                  0xe0178e05)
+NHSTUB(_sqrt,                               0xdffd619d) // TODO: different method signature, CalcSqrt, CalcSqrt_FW
+NHSTUB(apex2us,                             0xe01217c7)
+NHSTUB(exmem_ualloc,                        0xe00744f7)
+NHSTUB(exmem_ufree,                         0xe0074553)
+NHSTUB(reboot_fw_update,                    0xe039b733)
+NHSTUB(GetImageFolder,                      0xe02e3031)
+
+IGNORE(EngDrvRead)
+NHSTUB(EngDrvRead_FW,                       0xe05b1d13) // in wrappers
+
+NHSTUB(GetLensCurrentFocalLength,           0xe03f4561)
+NHSTUB(GetLensWideFocalLength,              0xe03f4573)
+
+NHSTUB(VTMLock,                             NULL_SUB)
+NHSTUB(VTMUnlock,                           NULL_SUB)
+DEF_CONST(uiprop_count,                     0x000000e8) // found 0x0xe025db4a
+FAKEDEF(focus_busy,                         1)
+FAKEDEF(zoom_busy,                          1)
+NHSTUB(MakeSDCardBootable,                  0xe036111b)
+DEF(current_fb_d,                           0x0000fd84) // found 0xe00c0ae2 (0xfd58 + 0x2c)
+DEF(recreview_hold,                         0x99e8) // found 0xe033c384 (0x9998 + 0x50)
+FAKEDEF(canon_shoot_menu_active,            1)
+DEF(simple_movie_status,                    0xb464) // found 0xe00dbb72 (0xb390 + 0x80 + 0x54)
+DEF(current_viewport_buffer,                0x00008870) // found 0xe03676da (0x8848 + 0x20 + 0x8)
+DEF(viewport_buffers,                       0) // TODO
```

Comment:
not sure about this ?

Reply:
DEFs go in stubs_min.S by convention. They point to data, or are constants. You'll need these when the port progresses.

Reply:
do you know how to get the viewport_buffers address ?

Reply:
You can look it up in a reference port's disasm and try locating it in your cam's disasm. Or, you can use a RAM dump that is large enough to contain those viewport buffers, use a utility on that dump to find the first of 4(?) buffers. When the first buffer's address is known, it can be searched (using a hex editor) in the ROM dump. One of the hits will be an array of 4 buffer addresses that have equal distance between them.
Anyway, I was curious and found it with the first method, the address is 0xE05E3E08.